crypto/src/secp256k1.rs

```diff
@@ -0,0 +1,353 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use crate::pubkey_bytes::PublicKeyBytes;
+use crate::traits::{
+    Authenticator, EncodeDecodeBase64, KeyPair, SigningKey, ToFromBytes, VerifyingKey,
+};
+use base64ct::{Base64, Encoding};
+use once_cell::sync::OnceCell;
+use serde::{de, Deserialize, Serialize};
+use signature::{Signature, Signer, Verifier};
+use std::fmt::{self, Debug, Display};
+
+pub const SECP256K1_PRIVATE_KEY_LENGTH: usize = 32;
+pub const SECP256K1_PUBLIC_KEY_LENGTH: usize = 33;
+pub const SECP256K1_SIGNATURE_LENGTH: usize = 64;
+
+#[readonly::make]
+#[derive(Debug, Clone)]
+pub struct Secp256k1PublicKey {
+    pub pubkey: k256::ecdsa::VerifyingKey,
+    pub bytes: OnceCell<[u8; SECP256K1_PUBLIC_KEY_LENGTH]>,
```

Comment:
I wonder if this is actually useful. Is there a way to implement `AsRef<[u8]>` for `k256::ecdsa::VerifyingKey`? If so, we would not need this `OnceCell`

Reply:
> I wonder if this is actually useful. Is there a way to implement `AsRef<[u8]>` for `k256::ecdsa::VerifyingKey`? If so, we would not need this `OnceCell`

there wasnt anyway to get the underlying bytes out of `k256::ecdsa::VerifyingKey` (its a private struct and no interface to retrieve this).

closest I can find, but this will not be AsRef
```
    /// Serialize this [`VerifyingKey`] as a SEC1-encoded bytestring
    /// (with point compression applied)
    pub fn to_bytes(&self) -> CompressedPoint {
        let mut result = [0u8; 33];
        result.copy_from_slice(EncodedPoint::from(self).as_ref());
        result
    }
```

Reply:
Entirely fair, then, to use the memo pattern.Thanks for looking into this.